spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/neo4j/Neo4jProperties.java

```diff
@@ -116,6 +123,13 @@ private void configure(Builder builder) {
 		if (this.username != null && this.password != null) {
 			builder.credentials(this.username, this.password);
 		}
+
+		builder.autoIndex(this.indexes.getAuto().name());
+
+		if (AutoIndexMode.DUMP == this.indexes.getAuto()) {
```

Comment:
Regarding the neo4j documentation (link in the PR description), the filename and directory configuration are here only for the dump mode.

But I can also avoid to make this configuration conditional.